src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -490,43 +493,138 @@ public Map<String, DirectiveDefinition> getDirectiveDefinitions() {
         return new LinkedHashMap<>(directiveDefinitions);
     }
 
+    /**
+     * Returns true if the registry has a type of the specified {@link TypeName}
+     *
+     * @param typeName the type name to check
+     *
+     * @return true if the registry has a type by that type name
+     */
     public boolean hasType(TypeName typeName) {
         String name = typeName.getName();
+        return hasType(name);
+    }
+
+    /**
+     * Returns true if the registry has a type of the specified name
+     *
+     * @param name the name to check
+     *
+     * @return true if the registry has a type by that name
+     */
+    public boolean hasType(String name) {
         return types.containsKey(name) || ScalarInfo.GRAPHQL_SPECIFICATION_SCALARS_DEFINITIONS.containsKey(name) || scalarTypes.containsKey(name) || objectTypeExtensions.containsKey(name);
     }
 
+    /**
+     * Returns am optional {@link TypeDefinition} of the specified type or {@link Optional#empty()}
+     *
+     * @param type the type to check
+     *
+     * @return an optional {@link TypeDefinition} or empty if it's not found
+     */
     public Optional<TypeDefinition> getType(Type type) {
-        String typeName = TypeInfo.typeInfo(type).getName();
-        return getType(typeName);
+        return getType(typeName(type));
```

Comment:
`TypeInfo.typeInfo(type).getName()` allocates a TypeInfo object in order to hust get a single attrbute of it - the name

So this is more light weight and we avoid an extra allocation